## Status
2026-06-06 19:07:48 UTC

### Reproducing
```
BuildAndTest.sh: line 55: 25255 Segmentation fault: 11  iccApplySearch -exportcfganddata config/cmykGraysEst.json Results/cmykGraysRef.txt 0 1 ICC/Spec380_10_730-D50_2deg.icc 3 ICC/Lab_float-D50_2deg.icc 3 ICC/CMYK_Hybrid_Profile.icc 10003 -INIT 3 ICC/Lab_float-D50_2deg.icc 1 ICC/Lab_float-D93_2deg-MAT.icc 1 ICC/Lab_float-F11_2deg-MAT.icc 1 ICC/Lab_float-IllumA_2deg-MAT.icc 1 > Results/cmykGraysEst.txt
Error: Process completed with exit code 139.
```

--- a/IccProfLib/IccCmm.cpp
+++ b/IccProfLib/IccCmm.cpp
@@ -7673,9 +7673,20 @@ IIccProfileConnectionConditions *CIccXformMpe::GetConnectionConditions() const
 */
 void CIccXformMpe::SetAppliedCC(IIccProfileConnectionConditions *pPCC)
 {
-  // Release any previously-owned applied PCC before reassigning below.
-  // SetAppliedCC can be invoked more than once per xform (e.g. from
-  // CIccCmm::SetLateBindingCC), and without this a prior combined PCC leaks.
+  // CIccCmm::SetLateBindingCC may run more than once (e.g. once per
+  // CIccCmmSearch::Begin candidate) and feeds an xform's own
+  // GetConnectionConditions() back into SetAppliedCC. If we are handed the
+  // connection conditions we already applied, there is nothing to do:
+  // re-wrapping would leak the current combined PCC, and because the new
+  // CIccCombinedConnectionConditions stores a pointer to it, freeing the old
+  // one would leave that wrapper dangling (use-after-free).
+  if (pPCC && pPCC == m_pAppliedPCC)
+    return;
+
+  // Otherwise we are replacing the applied PCC. Release any previously-owned
+  // one first so it does not leak across repeated calls. This is safe because
+  // the incoming pPCC does not alias it (that case returned above), so no
+  // freshly-built combined PCC can reference the object being freed.
   if (m_bDeleteAppliedPCC) {
     delete m_pAppliedPCC;
   }

